get_http.go

```diff
@@ -115,6 +144,20 @@ func (g *HttpGetter) GetFile(dst string, u *url.URL) error {
 	return err
 }
 
+func (g *HttpGetter) parseUrl(u *url.URL) (basicAuthUser, basicAuthPass string) {
```

Comment:
nit pick - this method name is pretty generic/broad but the actual implementation is pretty narrow, we're specifically grabbing the username/password, so a more descriptive name would be better